addon/mixins/cancelable.js

```diff
@@ -0,0 +1,50 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create({
+  resetInitialValues() {
+    let {
+      startDate,
+      endDate,
+      startMonth,
+      endMonth
+    } = this.getProperties('startDate', 'endDate', 'startMonth', 'endMonth');
+
+    this.setProperties({
+      initialStartDate: startDate.clone(),
+      initialEndDate: endDate.clone(),
+      initialStartMonth: startMonth.clone(),
+      initialEndMonth: endMonth.clone()
+    });
+  },
+
+  didInsertElement() {
+    this.resetInitialValues();
```

Comment:
Setting values in didInsertElement is deprecated: http://emberjs.com/deprecations/v1.x/#toc_modifying-a-property-within-code-didinsertelement-code
didReceiveAttrs is probably the more appropriate hook.